src/Dfc.CourseDirectory.WebV2/Features/Venues/DeleteVenue/DeleteVenue.cs

```diff
@@ -0,0 +1,232 @@
+        public IReadOnlyCollection<AffectedApprenticeshipViewModel> AffectedApprenticeships { get; set; }
+        public IReadOnlyCollection<AffectedTLevelViewModel> AffectedTLevels { get; set; }
+    }
+
+    public class ViewModel : Command
+    {
+        public string VenueName { get; set; }
+        public IReadOnlyCollection<string> AddressParts { get; set; }
+        public string PostCode { get; set; }
+    }
+
+    public class AffectedCourseViewModel
+    {
+        public Guid CourseId { get; set; }
+        public Guid CourseRunId { get; set; }
+        public string CourseName { get; set; }
+    }
+
+    public class AffectedApprenticeshipViewModel
+    {
+        public Guid ApprenticeshipId { get; set; }
+        public string ApprenticeshipName { get; set; }
+    }
+
+    public class AffectedTLevelViewModel
+    {
+        public Guid TLevelId { get; set; }
+        public string TLevelName { get; set; }
+    }
+
+    public class DeletedQuery : IRequest<DeletedViewModel>
+    {
+    }
+
+    public class DeletedViewModel
+    {
+        public string VenueName { get; set; }
+    }
+
+    public class Handler :
+        IRequestHandler<Query, OneOf<NotFound, ViewModel>>,
+        IRequestHandler<Command, OneOf<NotFound, ModelWithErrors<ViewModel>, Success>>,
+        IRequestHandler<DeletedQuery, DeletedViewModel>
+    {
+        private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+        private readonly ISqlQueryDispatcher _sqlQueryDispatcher;
+        private readonly IProviderOwnershipCache _providerOwnershipCache;
+        private readonly JourneyInstanceProvider _journeyInstanceProvider;
+
+        public Handler(
+            ICosmosDbQueryDispatcher cosmosDbQueryDispatcher,
+            ISqlQueryDispatcher sqlQueryDispatcher,
+            IProviderOwnershipCache providerOwnershipCache,
+            JourneyInstanceProvider journeyInstanceProvider)
+        {
+            _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+            _sqlQueryDispatcher = sqlQueryDispatcher;
+            _providerOwnershipCache = providerOwnershipCache;
+            _journeyInstanceProvider = journeyInstanceProvider;
+        }
+
+        public async Task<OneOf<NotFound, ViewModel>> Handle(Query request, CancellationToken cancellationToken)
+        {
+            var venue = await _cosmosDbQueryDispatcher.ExecuteQuery(new GetVenueById { VenueId = request.VenueId });
+
+            if (venue == null)
+            {
+                return new NotFound();
+            }
+
+            var getCourses = _cosmosDbQueryDispatcher.ExecuteQuery(new GetAllCoursesForProvider { ProviderUkprn = venue.Ukprn, CourseStatuses = CourseStatus.Live | CourseStatus.Pending | CourseStatus.BulkUploadPending | CourseStatus.BulkUploadReadyToGoLive | CourseStatus.APIPending | CourseStatus.APIReadyToGoLive | CourseStatus.MigrationPending | CourseStatus.MigrationReadyToGoLive });
```

Comment:
needs wrapping because github

Reply:
How did you come to that particular list of statuses to check for? Might be worth an explanatory comment.

Reply:
are we going to have efficiency or RU issues with this?

Reply:
> needs wrapping because github

I'll wrap it a little, but I won't wrap an `OR` enum value.

> How did you come to that particular list of statuses to check for? Might be worth an explanatory comment.

It's everything but `Archived` and `Deleted`. It's a horrible API but it doesn't require a comment to explain.

> are we going to have efficiency or RU issues with this?

Depends on how much load there is on the site (Cosmos) at any given time. I've discussed this with Cathie and we're happy to push forward on the understanding that it may fail, but the user receiving a 500 error is less bad then them being allowed to perform an action that affects data integrity. If it becomes a serious issue in the future we can limit the number of records returned and provide the user with appropriate messaging.

Reply:
> discussed this with Cathie
:+1: 

> "It's everything but Archived and Deleted"

That's knowledge that will be lost to time if it's not in a comment or commit message; and when/if another enum value is added it won't be obvious whether to extend the list with the new value without reverse engineering the original intent. In my view you can help a future soul by adding some more context.